src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -62,24 +65,46 @@ private HttpSystemProxy(WinInetProxyHelper proxyHelper, SafeWinHttpHandle sessio
                 {
                     // Process bypass list for manual setting.
                     string[] list = proxyHelper.ProxyBypass.Split(';');
-                    List<string> tmpList = new List<string>();
+                    _bypass = new List<Regex>();
 
                     foreach (string value in list)
                     {
                         string tmp = value.Trim();
+                        if (tmp.Length == 0)
+                        {
+                            continue;
+                        }
                         if (tmp == "<local>")
                         {
                             _bypassLocal = true;
                             continue;
                         }
-                        if (tmp.Length > 0)
+                        try
+                        {
+                            // Escape any special characters and unescape * to get wildcard pattern match.
+                            Regex re = new Regex(Regex.Escape(tmp).Replace("\\*", ".*?") + "$",
+                                            RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
+                            _bypass.Add(re);
+                        }
+                        catch
                         {
-                            tmpList.Add(tmp);
+                            if (NetEventSource.IsEnabled)
+                            {
+                                NetEventSource.Info(this, $"Failed to process {tmp} from bypass list.");
```

Comment:
>" [](start = 101, length = 1)

Add the exception message here.

Reply:
Note that interpolated strings like `$"Failed to process {tmp} from bypass list."` are expensive because they create a STringBuilder under the covers. If you expect this kind of tracing to be low overhead, you should not use interpolated strings. Instead simple append like
`NetEventSource.Info(this, "Failed to process " + tmp + " from bypass list.");`